Nim `devel` nightly `i386` build failing

### Description

https://github.com/nim-lang/nightlies/actions/runs/9948774194/job/27483951748

### Nim Version

https://github.com/nim-lang/Nim/commit/c11b3f3fc75156c1ce7ad31b631ed4196c3d5ba5

### Current Output

```text
/home/runner/work/nightlies/nightlies/nim/compiler/suggest.nim(38, 11) Hint: duplicate import of 'tables'; previous import here: /home/runner/work/nightlies/nightlies/nim/compiler/sigmatch.nim(18, 11) [DuplicateModuleImport]
.....................................................................................
Hint: mm: orc; opt: speed; options: -d:danger
171833 lines; 7.898s; 665.496MiB peakmem; proj: /home/runner/work/nightlies/nightlies/nim/compiler/nim.nim; out: /home/runner/.cache/nim/nim_r/nim.json [SuccessX]
bin/nim compile -f --incremental:off --compileonly --gen_mapping --cc:gcc --skipUserCfg --os:windows --cpu:amd64 -d:danger -d:gitHash:c11b3f3fc75156c1ce7ad31b631ed4196c3d5ba5 compiler/nim.nim
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/config/nim.cfg' [Conf]
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/config/config.nims' [Conf]
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/compiler/nim.cfg' [Conf]
.................................................................................................................................................................................................
/home/runner/work/nightlies/nightlies/nim/compiler/modulegraphs.nim(14, 11) Warning: imported and not used: 'algorithm' [UnusedImport]
.................
/home/runner/work/nightlies/nightlies/nim/compiler/semdata.nim(18, 17) Warning: imported and not used: 'astalgo' [UnusedImport]
.......................
/home/runner/work/nightlies/nightlies/nim/compiler/concepts.nim(14, 13) Warning: imported and not used: 'astalgo' [UnusedImport]
.
/home/runner/work/nightlies/nightlies/nim/compiler/suggest.nim(38, 11) Hint: duplicate import of 'tables'; previous import here: /home/runner/work/nightlies/nightlies/nim/compiler/sigmatch.nim(18, 11) [DuplicateModuleImport]
.....................................................................................
Hint: mm: orc; opt: speed; options: -d:danger
171833 lines; 7.797s; 665.688MiB peakmem; proj: /home/runner/work/nightlies/nightlies/nim/compiler/nim.nim; out: /home/runner/.cache/nim/nim_r/nim.json [SuccessX]
bin/nim compile -f --incremental:off --compileonly --gen_mapping --cc:gcc --skipUserCfg --os:linux --cpu:i386 -d:danger -d:gitHash:c11b3f3fc75156c1ce7ad31b631ed4196c3d5ba5 compiler/nim.nim
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/config/nim.cfg' [Conf]
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/config/config.nims' [Conf]
Hint: used config file '/home/runner/work/nightlies/nightlies/nim/compiler/nim.cfg' [Conf]
...........................................................................................................................................................
/home/runner/work/nightlies/nightlies/nim/lib/pure/osproc.nim(1409, 34) Error: type mismatch: got 'int64' for 'ns' but expected 'int'
Error: call to nim compiler failed
Error: execution of an external program failed: '/home/runner/work/nightlies/nightlies/nim/tools/niminst/niminst --var:version=2.1.9 --var:mingw=none csource --main:compiler/nim.nim compiler/installer.ini -d:danger -d:gitHash:c11b3f3fc75156c1ce7ad31b631ed4196c3d5ba5'
FAILURE
Error: Process completed with exit code 1.
```


### Expected Output

```text
Nightly builds
```


### Possible Solution

_No response_

### Additional Information

_No response_

--- a/lib/pure/osproc.nim
+++ b/lib/pure/osproc.nim
@@ -1406,7 +1406,7 @@ elif not defined(useNimRtl):
                 waitSpec: TimeSpec
                 unused: Timespec
               waitSpec.tv_sec = posix.Time(secs)
-              waitSpec.tv_nsec = ns 
+              waitSpec.tv_nsec = int ns 
               discard posix.clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, waitSpec, unused)
               let remaining = deadline - getMonoTime()
               delay = min([delay * 2, remaining, maxWait])

--- a/lib/pure/osproc.nim
+++ b/lib/pure/osproc.nim
@@ -1406,7 +1406,7 @@ elif not defined(useNimRtl):
                 waitSpec: TimeSpec
                 unused: Timespec
               waitSpec.tv_sec = posix.Time(secs)
-              waitSpec.tv_nsec = int ns 
+              waitSpec.tv_nsec = clong ns 
               discard posix.clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, waitSpec, unused)
               let remaining = deadline - getMonoTime()
               delay = min([delay * 2, remaining, maxWait])

